도서관 세미나룸 예약 현황 요청 시 MediaType 불일치로 응답 디코딩이 불가능한 경우에 대한 예외 처리 추가

### 🐞 설명

[문제]
Open Feign으로 도서관 세미나룸의 예약 현황 요청하는게 잘 되다가 갑자기 MediaType이 application/octet-stream으로 응답이 와서 (json 타입이 아니라서) 서버 내에서 dto로 변환이 불가능 해 에러가 발생하였습니다.

[원인]
정확한 이유는 모르겠으나, 현재 토큰이 만료되기 전까지는 동일한 토큰으로 계속 api를 호출하도록 하고 있는데, 너무 많은 요청을 보내서 해당 토큰으로 들어온 요청을 블락하는 것은 아닐까 하는 가정을 했습니다.

[해결방안]
Open Feign으로 도서관 api 호출 과정에서 예외가 발생할 경우, 사용하던 토큰을 버리고 도서관에 재로그인 후 새로운 토큰을 받아서 활용하도록 했습니다.

### 📄 버그 로그

```
feign.codec.DecodeException: Could not extract response: no suitable HttpMessageConverter found for response type [class com.yourssu.openssupot.campus.domain.domain.oasis.SeminarRoomsResponse] and content type [application/octet-stream]
	at feign.InvocationContext.decode(InvocationContext.java:118) ~[feign-core-13.3.jar!/:na]
	at feign.InvocationContext.proceed(InvocationContext.java:88) ~[feign-core-13.3.jar!/:na]
	at feign.ResponseHandler.handleResponse(ResponseHandler.java:63) ~[feign-core-13.3.jar!/:na]
	at feign.SynchronousMethodHandler.executeAndDecode(SynchronousMethodHandler.java:114) ~[feign-core-13.3.jar!/:na]
	at feign.SynchronousMethodHandler.invoke(SynchronousMethodHandler.java:70) ~[feign-core-13.3.jar!/:na]
	at feign.ReflectiveFeign$FeignInvocationHandler.invoke(ReflectiveFeign.java:99) ~[feign-core-13.3.jar!/:na]
	at jdk.proxy2/jdk.proxy2.$Proxy152.getSeminarRooms(Unknown Source) ~[na:na]
	at com.yourssu.openssupot.campus.domain.domain.oasis.OasisClient$DefaultImpls.getSeminarRooms$default(OasisClient.kt:17) ~[!/:0.0.1-SNAPSHOT]
	at com.yourssu.openssupot.campus.domain.domain.oasis.OasisReader.getSeminarRoomsResponse(OasisReader.kt:32) ~[!/:0.0.1-SNAPSHOT]
	at com.yourssu.openssupot.campus.domain.domain.oasis.OasisReader.getSeminarRoomsResponse$default(OasisReader.kt:28) ~[!/:0.0.1-SNAPSHOT]
	at com.yourssu.openssupot.campus.domain.domain.oasis.OasisReader.getAllByDate(OasisReader.kt:15) ~[!/:0.0.1-SNAPSHOT]
	at com.yourssu.openssupot.campus.domain.domain.CampusSpaceService.readFromOasis(CampusSpaceService.kt:30) ~[!/:0.0.1-SNAPSHOT]
	at com.yourssu.openssupot.campus.domain.domain.CampusSpaceService.readAllByTimeRange(CampusSpaceService.kt:20) ~[!/:0.0.1-SNAPSHOT]
	at com.yourssu.openssupot.campus.application.domain.CampusSpaceController.readAll(CampusSpaceController.kt:20) ~[!/:0.0.1-SNAPSHOT]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:580) ~[na:na]
	at kotlin.reflect.jvm.internal.calls.CallerImpl$Method.callMethod(CallerImpl.kt:97) ~[kotlin-reflect-1.9.25.jar!/:1.9.25-release-852]
	at kotlin.reflect.jvm.internal.calls.CallerImpl$Method$Instance.call(CallerImpl.kt:113) ~[kotlin-reflect-1.9.25.jar!/:1.9.25-release-852]
	at kotlin.reflect.jvm.internal.KCallableImpl.callDefaultMethod$kotlin_reflection(KCallableImpl.kt:207) ~[kotlin-reflect-1.9.25.jar!/:1.9.25-release-852]
	at kotlin.reflect.jvm.internal.KCallableImpl.callBy(KCallableImpl.kt:112) ~[kotlin-reflect-1.9.25.jar!/:1.9.25-release-852]
	at org.springframework.web.method.support.InvocableHandlerMethod$KotlinDelegate.invokeFunction(InvocableHandlerMethod.java:334) ~[spring-web-6.1.14.jar!/:6.1.14]
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:252) ~[spring-web-6.1.14.jar!/:6.1.14]
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:188) ~[spring-web-6.1.14.jar!/:6.1.14]
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:118) ~[spring-webmvc-6.1.14.jar!/:6.1.14]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:926) ~[spring-webmvc-6.1.14.jar!/:6.1.14]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:831) ~[spring-webmvc-6.1.14.jar!/:6.1.14]
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87) ~[spring-webmvc-6.1.14.jar!/:6.1.14]
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1089) ~[spring-webmvc-6.1.14.jar!/:6.1.14]
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:979) ~[spring-webmvc-6.1.14.jar!/:6.1.14]
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1014) ~[spring-webmvc-6.1.14.jar!/:6.1.14]
	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:903) ~[spring-webmvc-6.1.14.jar!/:6.1.14]
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:564) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:885) ~[spring-webmvc-6.1.14.jar!/:6.1.14]
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:658) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:195) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51) ~[tomcat-embed-websocket-10.1.31.jar!/:na]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100) ~[spring-web-6.1.14.jar!/:6.1.14]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.1.14.jar!/:6.1.14]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93) ~[spring-web-6.1.14.jar!/:6.1.14]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.1.14.jar!/:6.1.14]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201) ~[spring-web-6.1.14.jar!/:6.1.14]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.1.14.jar!/:6.1.14]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:167) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:483) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:384) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:905) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1741) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at java.base/java.lang.Thread.run(Thread.java:1583) ~[na:na]
Caused by: org.springframework.web.client.UnknownContentTypeException: Could not extract response: no suitable HttpMessageConverter found for response type [class com.yourssu.openssupot.campus.domain.domain.oasis.SeminarRoomsResponse] and content type [application/octet-stream]
	at org.springframework.web.client.HttpMessageConverterExtractor.extractData(HttpMessageConverterExtractor.java:125) ~[spring-web-6.1.14.jar!/:6.1.14]
	at org.springframework.cloud.openfeign.support.SpringDecoder.decode(SpringDecoder.java:71) ~[spring-cloud-openfeign-core-4.1.3.jar!/:4.1.3]
	at org.springframework.cloud.openfeign.support.ResponseEntityDecoder.decode(ResponseEntityDecoder.java:62) ~[spring-cloud-openfeign-core-4.1.3.jar!/:4.1.3]
	at feign.optionals.OptionalDecoder.decode(OptionalDecoder.java:36) ~[feign-core-13.3.jar!/:na]
	at feign.InvocationContext.decode(InvocationContext.java:114) ~[feign-core-13.3.jar!/:na]
	... 66 common frames omitted

2024-11-24T18:56:30.253+09:00 TRACE 40486 --- [OpenSSUpot] [io-8080-exec-10] o.s.b.w.s.f.OrderedRequestContextFilter  : Cleared thread-bound request context: org.apache.catalina.connector.RequestFacade@3ff4fb5d
2024-11-24T18:56:30.254+09:00 ERROR 40486 --- [OpenSSUpot] [io-8080-exec-10] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed: feign.codec.DecodeException: Could not extract response: no suitable HttpMessageConverter found for response type [class com.yourssu.openssupot.campus.domain.domain.oasis.SeminarRoomsResponse] and content type [application/octet-stream]] with root cause

org.springframework.web.client.UnknownContentTypeException: Could not extract response: no suitable HttpMessageConverter found for response type [class com.yourssu.openssupot.campus.domain.domain.oasis.SeminarRoomsResponse] and content type [application/octet-stream]
	at org.springframework.web.client.HttpMessageConverterExtractor.extractData(HttpMessageConverterExtractor.java:125) ~[spring-web-6.1.14.jar!/:6.1.14]
	at org.springframework.cloud.openfeign.support.SpringDecoder.decode(SpringDecoder.java:71) ~[spring-cloud-openfeign-core-4.1.3.jar!/:4.1.3]
	at org.springframework.cloud.openfeign.support.ResponseEntityDecoder.decode(ResponseEntityDecoder.java:62) ~[spring-cloud-openfeign-core-4.1.3.jar!/:4.1.3]
	at feign.optionals.OptionalDecoder.decode(OptionalDecoder.java:36) ~[feign-core-13.3.jar!/:na]
	at feign.InvocationContext.decode(InvocationContext.java:114) ~[feign-core-13.3.jar!/:na]
	at feign.InvocationContext.proceed(InvocationContext.java:88) ~[feign-core-13.3.jar!/:na]
	at feign.ResponseHandler.handleResponse(ResponseHandler.java:63) ~[feign-core-13.3.jar!/:na]
	at feign.SynchronousMethodHandler.executeAndDecode(SynchronousMethodHandler.java:114) ~[feign-core-13.3.jar!/:na]
	at feign.SynchronousMethodHandler.invoke(SynchronousMethodHandler.java:70) ~[feign-core-13.3.jar!/:na]
	at feign.ReflectiveFeign$FeignInvocationHandler.invoke(ReflectiveFeign.java:99) ~[feign-core-13.3.jar!/:na]
	at jdk.proxy2/jdk.proxy2.$Proxy152.getSeminarRooms(Unknown Source) ~[na:na]
	at com.yourssu.openssupot.campus.domain.domain.oasis.OasisClient$DefaultImpls.getSeminarRooms$default(OasisClient.kt:17) ~[!/:0.0.1-SNAPSHOT]
	at com.yourssu.openssupot.campus.domain.domain.oasis.OasisReader.getSeminarRoomsResponse(OasisReader.kt:32) ~[!/:0.0.1-SNAPSHOT]
	at com.yourssu.openssupot.campus.domain.domain.oasis.OasisReader.getSeminarRoomsResponse$default(OasisReader.kt:28) ~[!/:0.0.1-SNAPSHOT]
	at com.yourssu.openssupot.campus.domain.domain.oasis.OasisReader.getAllByDate(OasisReader.kt:15) ~[!/:0.0.1-SNAPSHOT]
	at com.yourssu.openssupot.campus.domain.domain.CampusSpaceService.readFromOasis(CampusSpaceService.kt:30) ~[!/:0.0.1-SNAPSHOT]
	at com.yourssu.openssupot.campus.domain.domain.CampusSpaceService.readAllByTimeRange(CampusSpaceService.kt:20) ~[!/:0.0.1-SNAPSHOT]
	at com.yourssu.openssupot.campus.application.domain.CampusSpaceController.readAll(CampusSpaceController.kt:20) ~[!/:0.0.1-SNAPSHOT]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:580) ~[na:na]
	at kotlin.reflect.jvm.internal.calls.CallerImpl$Method.callMethod(CallerImpl.kt:97) ~[kotlin-reflect-1.9.25.jar!/:1.9.25-release-852]
	at kotlin.reflect.jvm.internal.calls.CallerImpl$Method$Instance.call(CallerImpl.kt:113) ~[kotlin-reflect-1.9.25.jar!/:1.9.25-release-852]
	at kotlin.reflect.jvm.internal.KCallableImpl.callDefaultMethod$kotlin_reflection(KCallableImpl.kt:207) ~[kotlin-reflect-1.9.25.jar!/:1.9.25-release-852]
	at kotlin.reflect.jvm.internal.KCallableImpl.callBy(KCallableImpl.kt:112) ~[kotlin-reflect-1.9.25.jar!/:1.9.25-release-852]
	at org.springframework.web.method.support.InvocableHandlerMethod$KotlinDelegate.invokeFunction(InvocableHandlerMethod.java:334) ~[spring-web-6.1.14.jar!/:6.1.14]
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:252) ~[spring-web-6.1.14.jar!/:6.1.14]
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:188) ~[spring-web-6.1.14.jar!/:6.1.14]
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:118) ~[spring-webmvc-6.1.14.jar!/:6.1.14]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:926) ~[spring-webmvc-6.1.14.jar!/:6.1.14]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:831) ~[spring-webmvc-6.1.14.jar!/:6.1.14]
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87) ~[spring-webmvc-6.1.14.jar!/:6.1.14]
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1089) ~[spring-webmvc-6.1.14.jar!/:6.1.14]
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:979) ~[spring-webmvc-6.1.14.jar!/:6.1.14]
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1014) ~[spring-webmvc-6.1.14.jar!/:6.1.14]
	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:903) ~[spring-webmvc-6.1.14.jar!/:6.1.14]
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:564) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:885) ~[spring-webmvc-6.1.14.jar!/:6.1.14]
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:658) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:195) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51) ~[tomcat-embed-websocket-10.1.31.jar!/:na]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100) ~[spring-web-6.1.14.jar!/:6.1.14]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.1.14.jar!/:6.1.14]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93) ~[spring-web-6.1.14.jar!/:6.1.14]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.1.14.jar!/:6.1.14]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201) ~[spring-web-6.1.14.jar!/:6.1.14]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.1.14.jar!/:6.1.14]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:167) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:483) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:384) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:905) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1741) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63) ~[tomcat-embed-core-10.1.31.jar!/:na]
	at java.base/java.lang.Thread.run(Thread.java:1583) ~[na:na]
```

### 🌏 버그 발생 환경

.

### 🙋🏻 버그 재현 방법

_No response_

### 🙋🏻 참고 자료

_No response_

--- a/src/main/kotlin/com/yourssu/openssupot/campus/domain/domain/oasis/OasisClient.kt
+++ b/src/main/kotlin/com/yourssu/openssupot/campus/domain/domain/oasis/OasisClient.kt
@@ -16,15 +16,7 @@ interface OasisClient {
     @GetMapping("/pyxis-api/1/api/rooms")
     fun getSeminarRooms(
         @RequestHeader("pyxis-auth-token") accessToken: String,
-        @RequestParam("roomTypeId") roomTypeId: Long = 1L,
-        @RequestParam("smufMethodCode") smufMethodCode: String = "PC",
-        @RequestParam("hopeDate") hopeDate: String,
-    ): SeminarRoomsResponse
-
-    @GetMapping("/pyxis-api/1/api/rooms")
-    fun getOpenSeminarRooms(
-        @RequestHeader("pyxis-auth-token") accessToken: String,
-        @RequestParam("roomTypeId") roomTypeId: Long = 5L,
+        @RequestParam("roomTypeId") roomTypeId: Long,
         @RequestParam("smufMethodCode") smufMethodCode: String = "PC",
         @RequestParam("hopeDate") hopeDate: String,
     ): SeminarRoomsResponse

--- a/src/main/kotlin/com/yourssu/openssupot/campus/domain/domain/oasis/OasisReader.kt
+++ b/src/main/kotlin/com/yourssu/openssupot/campus/domain/domain/oasis/OasisReader.kt
@@ -25,22 +25,33 @@ class OasisReader(
     private fun getSeminarRoomsResponse(
         roomTypeId: Int,
         date: LocalDate,
-        isRecursive: Boolean = false,
     ): SeminarRoomsResponse {
-        val seminarRoomsResponse: SeminarRoomsResponse = oasisClient.getSeminarRooms(
+        try {
+            val seminarRoomResponse: SeminarRoomsResponse = getFromOasis(roomTypeId, date)
+            if (!seminarRoomResponse.success) {
+                oasisTokenProvider.invalidateAccessToken()
+                oasisTokenProvider.fetchNewAccessToken()
+
+                return getFromOasis(roomTypeId, date)
+            }
+
+            return seminarRoomResponse
+        } catch (e: Exception) {
+            oasisTokenProvider.invalidateAccessToken()
+            oasisTokenProvider.fetchNewAccessToken()
+
+            return getFromOasis(roomTypeId, date)
+        }
+    }
+
+    private fun getFromOasis(
+        roomTypeId: Int,
+        date: LocalDate,
+    ): SeminarRoomsResponse {
+        return oasisClient.getSeminarRooms(
             accessToken = oasisTokenProvider.getAccessToken(),
             roomTypeId = roomTypeId.toLong(),
             hopeDate = date.toString(),
         )
-        if (!seminarRoomsResponse.success) {
-            check(!isRecursive) { "Failed to get seminar rooms" }
-
-            oasisTokenProvider.invalidateAccessToken()
-            oasisTokenProvider.fetchNewAccessToken()
-
-            return getSeminarRoomsResponse(roomTypeId, date, true)
-        }
-
-        return seminarRoomsResponse
     }
 }

--- a/src/main/kotlin/com/yourssu/openssupot/campus/domain/domain/oasis/OasisReader.kt
+++ b/src/main/kotlin/com/yourssu/openssupot/campus/domain/domain/oasis/OasisReader.kt
@@ -12,25 +12,24 @@ class OasisReader(
     fun getAllByDate(date: LocalDate): List<SeminarRoom> {
         val seminarRooms = mutableListOf<SeminarRoom>()
 
-        val seminarRoomsResponse: SeminarRoomsResponse = getSeminarRoomsResponse(date)
-        seminarRooms.addAll(seminarRoomsResponse.data.list.map {
-            SeminarRoom.from(it, RoomType.SEMINAR_ROOM)
-        })
-
-        val openSeminarRoomsResponse: SeminarRoomsResponse = getOpenSeminarRoomsResponse(date)
-        seminarRooms.addAll(openSeminarRoomsResponse.data.list.map {
-            SeminarRoom.from(it, RoomType.OPEN_SEMINAR_ROOM)
-        })
+        for (roomType in RoomType.entries) {
+            val seminarRoomsResponse: SeminarRoomsResponse = getSeminarRoomsResponse(roomType.id, date)
+            seminarRooms.addAll(seminarRoomsResponse.data.list.map {
+                SeminarRoom.from(it, roomType)
+            })
+        }
 
         return seminarRooms
     }
 
     private fun getSeminarRoomsResponse(
+        roomTypeId: Int,
         date: LocalDate,
         isRecursive: Boolean = false,
     ): SeminarRoomsResponse {
         val seminarRoomsResponse: SeminarRoomsResponse = oasisClient.getSeminarRooms(
             accessToken = oasisTokenProvider.getAccessToken(),
+            roomTypeId = roomTypeId.toLong(),
             hopeDate = date.toString(),
         )
         if (!seminarRoomsResponse.success) {
@@ -39,29 +38,9 @@ class OasisReader(
             oasisTokenProvider.invalidateAccessToken()
             oasisTokenProvider.fetchNewAccessToken()
 
-            return getSeminarRoomsResponse(date, true)
+            return getSeminarRoomsResponse(roomTypeId, date, true)
         }
 
         return seminarRoomsResponse
-    }
-
-    private fun getOpenSeminarRoomsResponse(
-        date: LocalDate,
-        isRecursive: Boolean = false,
-    ): SeminarRoomsResponse {
-        val openSeminarRoomsResponse: SeminarRoomsResponse = oasisClient.getOpenSeminarRooms(
-            accessToken = oasisTokenProvider.getAccessToken(),
-            hopeDate = date.toString(),
-        )
-        if (!openSeminarRoomsResponse.success) {
-            check(!isRecursive) { "Failed to get open seminar rooms" }
-
-            oasisTokenProvider.invalidateAccessToken()
-            oasisTokenProvider.fetchNewAccessToken()
-
-            return getOpenSeminarRoomsResponse(date, true)
-        }
-
-        return openSeminarRoomsResponse
     }
 }

